external-import/ransomwarelive/src/ransomwarelive/ransom_conn.py

```
@@ -0,0 +1,675 @@
+            return None
+        except Exception as e:
+            self.helper.connector_logger.error(
+                "Error fetching location", {"country": country, "error": e}
+            )
+            return None
+
+    def sector_fetcher(self, sector: str):
+        """
+        Fetch the sector object related to param by searching with conditions:
+            - entity_type is "sector"
+            - name is egual to sector string OR x_opencti_aliases is egual to sector string
+        :param sector: sector in string
+        :return: sector id or None
+        """
+        if sector == "":
+            return None
+
+        sector_out = None
+        rubbish = [" and ", " or ", " ", ";"]
+        for item in rubbish:
+            sector = " ".join(sector.split(item))
+
+        sector_out = self.helper.api.identity.read(
+            filters={
+                "mode": "and",
+                "filters": [
+                    {
+                        "key": "entity_type",
+                        "values": ["Sector"],
+                        "operator": "eq",
+                    },
+                ],
+                "filterGroups": [
+                    {
+                        "mode": "or",
+                        "filters": [
+                            {
+                                "key": "name",
+                                "values": sector,
+                                "operator": "eq",
+                            },
+                            {
+                                "key": "x_opencti_aliases",
+                                "values": sector,
+                                "operator": "eq",
+                            },
+                        ],
+                        "filterGroups": [],
+                    }
+                ],
+            },
+        )
+
+        if sector_out and sector_out.get("standard_id").startswith("identity--"):
+            return sector_out.get("standard_id")
+
```

Actually I think I was wrong when I told you to let errors propagate here, it may depend. If the sector is not mandatory to build the bundle, logging the error might be just fine (just like in `location_fetcher`). WDYT?

---

up